dotcom-rendering/src/model/enhanceCollections.ts

```diff
@@ -26,6 +48,12 @@ export const enhanceCollections = (
 		const containerPalette = decideContainerPalette(
 			collection.config.metadata?.map((meta) => meta.type),
 		);
+		const allBranding = getBrandingFromCards(
+			[...collection.curated, ...collection.curated],
+			editionId,
+		);
+		const allCardsHaveBranding = allCardsHaveSponsors(allBranding);
```

Comment:
If the branding isn't used unless _all_ the cards have branding, perhaps `getBrandingFromCards` can return just the brandings (e.g `Branding[]` rather than `(Branding | Undefined)[]`) - by filtering on undefined.

Then you could simplify `allCardsHaveBranding = allBranding.length === allCards.length` (you'd need to define `const allCards = [...collection.curated, ...collection.curated]`

Reply:
There’s [`isNonNullable` from `@guardian/lib`](https://github.com/guardian/csnx/blob/a64bbc6d1fced5ced51c9295dc53163706855247/libs/%40guardian/libs/src/isNonNullable/isNonNullable.ts) which may help solve this?

```ts
const cardsWithBranding: Branding[] = allCards
  .map(() => { /* do stuff */ })
  .filter(isNonNullable);
```